developer-docs-site/docs/nodes/validator-node/operator/delegation-pool-operations.md

```diff
@@ -155,7 +155,20 @@ Alternatively, you can use Aptos CLI to call View functions.
 
 ```bash
  aptos move view [OPTIONS] --function-id <FUNCTION_ID>
-
 ```
 
 To discover the available options and the process for making an `aptos move view` call, access the help information with `aptos move view --help`. This will display the required arguments for invoking the view functions.
+
+
+## Compute delegation pool rewards earned
+Use this formula to calculate *rewards earned* for `active` and `pending_inactive` staking. This formula assumes that different stake operations such as `unlock` and `reactivate` take out the *principals* first and then *rewards*. Therefore, *rewards earned* may vary based upon how the formula you use is constructed:
+
+1. Get the amount of `active` and `pending_inactive` staking from the [`get_stake`](https://github.com/aptos-labs/aptos-core/blob/ed63ab756cda61439287304ed89bbb156fcbeaed/aptos-move/framework/aptos-framework/sources/delegation_pool.move#L321) view function.
+
+2. Calculate principal:
+    - "active principal" = **AddStakeEvent** - **UnlockStakeEvent** + **ReactivateStakeEvent**. If at any point during the iteration, "active principal" < 0, reset to 0. Negative principal could happen when the amount users `unlock` or `reactivate` include rewards earned from staking.
+    - "pending inactive principal" = **UnlockStakeEvent** - **ReactivateStakeEvent**. If at any point during the iteration, "pending inactive principal" < 0, reset to 0. Negative principal could happen when the amount users `unlock` or `reactivate` include rewards earned from staking.
```

Comment:
This doesn't quite make sense to me. If I have 10 APT of principal and 50 APT of rewards. If UnlockStakeEvent says I'm unlocking 50 APT, this would show 50 APT of pending inactive principal?

Reply:
we don't display principal anywhere, it's just a concept that makes easy to understand when computing rewards. in your example, if we stake 10 apt, and earn 50 apt in rewards, we will have 10 apt `AddStakeEvent`. when we unlock 50 apt, we will have 50 apt `UnlockStakeEvent`. then for pending inactive rewards, it will be pending_inactive as 50 
 - pending_inactive_principal as 50 = 0.

Reply:
I see. I'd suggest adding a simple example here to make this easier to understand